### PR TITLE
update as_pymongo method from main mongoengine project

### PR DESCRIPTION
**Pull request by danielgeng**
This PR brings the pymongo changes from https://github.com/MongoEngine/mongoengine/pull/1549

Currently, `.only()` and `.as_pymongo()` don't work well together for fields where the underlying data is a dict (e.g. EmbeddedDocumentField, RPFileField, etc) since you need to specify all the fields within the embedded field that you want. 

For example, `Employee.objects.only('id', 'name').as_pymongo()` will return something like:
```
{'_id': ObjectId('60a58a1306077651fdc5b372'), 'name': {}}
```
since there aren't any nested fields of `name` specified.

There isn't an easy way to return all the nested fields of name without explicitly specifying all the nested fields within `.only()`. In the case of `Employee.name`, if you want all fields within name, the query would look something like `Employee.objects.only('id', 'name.first', 'name.middle', 'name.last', 'name.title', 'name.suffix', ...).as_pymongo()` which is messy.

For `EmbeddedDocumentField`, the fields could technically be retrieved by doing something like `Employee.name.document_type._fields`. However, I wasn't able to find a similar way to do this for `RPFileField` fields because `savedFileName` isn't an actual field, so it breaks `.only()` -- e.g. `SignedDocument.objects_across_company.only('finalPdfUrl.savedFileName')`

After this change, the result of  `Employee.objects.only('id', 'name').as_pymongo()` will be:

```
{
    '_id': ObjectId('60a58a1306077651fdc5b372'), 
    'name': {
        'title': 'mr.', 
        'first': 'test', 
        'last': 'lol', 
        'nickname': 'abc', 
        'full': 'mr. test lol (abc)'
    }
}
```

The reason that I want to make this change is because I want to use pymongo for speeding up mongo data fetching in RQL and since there are cases where we want all fields of some dict based field. Omitting `.only()` isn't a viable option since it greatly increase the computation time in many cases (we typically only fetch 1 or 2 fields).

[ci full] passed for rippling-main: https://github.com/Rippling/rippling-main/pull/70903

### CODE REVIEW

File: mongoengine/queryset/base.py [hunk 1]
```diff
@@ -1082,12 +1081,11 @@ def as_pymongo(self, coerce_types=False):
         """Instead of returning Document instances, return raw values from
```
Comment (Chipe1) [inline mongoengine/queryset/base.py:1081]: `coerce_types` kwarg should be removed

File: tests/queryset/queryset.py [hunk 2]
```diff
@@ -4261,21 +4261,44 @@ class Doc(Document):
...
             name = StringField()
             age = IntField()
             price = DecimalField()
+            last_login = EmbeddedDocumentField(LastLogin)
 
         User.drop_collection()
-        User(name="Bob Dole", age=89, price=Decimal('1.11')).save()
-        User(name="Barack Obama", age=51, price=Decimal('2.22')).save()
+
+        User.objects.create(name="Bob Dole", age=89, price=Decimal('1.11'))
```
Comment (Chipe1) [inline tests/queryset/queryset.py:4279]: We don't have pipelines to run tests on this repo. Can you add a test in rippling-main

**Review by Chipe1 — APPROVED** [review]

**Comment by Chipe1** [comment]
@danielgeng You use pymongo directly if performance is a concern - it has a cleaner API and is well documented+maintained. `Employee._get_collection()` returns a [pymongo collection](https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html)

File: mongoengine/queryset/base.py (hunk 1 above)
Comment (danielgeng) [reply mongoengine/queryset/base.py:1081]: good catch, i'll make this change

File: tests/queryset/queryset.py (hunk 2 above)
Comment (danielgeng) [reply tests/queryset/queryset.py:4279]: sure

**Comment by danielgeng** [comment]
@Chipe1 if i understand correctly, if we use pymongo directly, we won't use `RPReadPreferenceContext` so we would have to specify `read_preference` manually in `_get_collection()`

we have a lot of usages of the read_preference decorators in RQL, so i thought it would be more maintainable to do it this way

**Comment by Chipe1** [comment]
> we would have to specify `read_preference` manually in `_get_collection()`
`RPDocumentBase._get_collection()` respects RPReadPreferenceContext and returns collection with read preference set

**Pull request merged**